### PR TITLE
feat(ci): publish linux/arm64 image

### LINKED ISSUE #5704: Docker Multi-platform Images -- linux/arm64 (and amd64)
**Is your feature request related to a problem? Please describe.**
Currently there is no linux/arm64 docker image for semgrep.

Currently Docker Hub only has a linux/amd64 image and PyPi has Linux and MacOS (x64 and M1).

**Describe the solution you'd like**
It would be nice to have a multi-platform docker image with a linux/arm64 in addition to linux/amd64.

This would possibly allow faster execution in docker on M1 Macs as the it will avoid emulation.

**Describe alternatives you've considered**
Install natively via PyPi, or Homebrew etc. But sometimes it is nice to use Docker.
Use different hardware.

**Use case**
Running semgrep in docker on Apple M1 and other ARM64 architectures using ARM64 natively rather than emulation of x64

**Additional context**
Related to https://github.com/returntocorp/semgrep/issues/2252

### PR DESCRIPTION
**Pull request by wilsonehusin**
ref: https://github.com/docker/build-push-action/blob/42863b1282f7046f6c1ec9222965bbceafae3bb8/docs/advanced/multi-platform.md

PR checklist:

- [x] Documentation is up-to-date
- [x] `changelog.d/<issue>.<type>` is a file with the _what_, _why_, and _how_ of the change.
  - \<issue> is `pa-312` (Linear ticket), `gh-1234` (GitHub issue), or `new-gizmo` (unique semantic name)
  - \<type> is `added`, `changed`, `fixed`, or `infra`.
- [x] Change has no security implications (otherwise, ping security team)

If you're unsure on any of this, please see the [Contribution guidelines](https://semgrep.dev/docs/contributing/how-to-contribute/)!

Edit: 

Closes #5704

### CODE REVIEW

**Comment by ievans** [comment]
Closes https://github.com/returntocorp/semgrep/issues/5704

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -88,13 +88,17 @@ jobs:
     runs-on: ubuntu-latest
     if: ${{ !contains(github.ref,'-test-release') }}
     steps:
+      - name: Set up QEMU # required for multiarch
+        uses: docker/setup-qemu-action@v2
+
```
Comment (brendongo) [inline .github/workflows/release.yml]:
```suggestion
```
To appease pre-commit

Comment (wilsonehusin) [reply .github/workflows/release.yml]: 😆 i thought i fixed it, thank you!

**Review by brendongo — APPROVED** [review]

**Comment by brendongo** [comment]
Thanks @wilsonehusin

**Comment by wilsonehusin** [comment]
🤔 I'm looking at Docker Hub and I found `301cf75f39f5444e95d700aec04abbc5a2b403d6f454dedc5843efccf40007b4` digest tagged `develop` but somehow the arm64 build isn't there

Edit: looks like I missed the builds / pushes for current HEAD — I'm addressing that in #5747

**Pull request merged**